lib/asset/assetmultipart.js

```diff
@@ -39,13 +40,13 @@ class AssetMultipart {
      */
     constructor(targetUrls, minPartSize, maxPartSize, headers, completeUrl, uploadToken) {
         if (!Array.isArray(targetUrls) || (targetUrls.length === 0)) {
-            throw Error(`Invalid target urls: ${targetUrls}`);
+            throw new IllegalArgumentError("Invalid target urls", targetUrls);
         } else if (!isValidNumber(minPartSize) || (minPartSize <= 0)) {
-            throw Error(`Invalid minPartSize: ${minPartSize}`);
+            throw new IllegalArgumentError("Invalid minPartSize", minPartSize);
         } else if (!isValidNumber(maxPartSize) || (maxPartSize < minPartSize)) {
-            throw Error(`Invalid maxPartSize: ${maxPartSize}, minPartSize: ${minPartSize}`);
+            throw new IllegalArgumentError(`Invalid maxPartSize (minPartSize=${minPartSize})`, maxPartSize);
         } else if (completeUrl && !(completeUrl instanceof URL) && (typeof completeUrl !== "string")) {
-            throw Error(`Invalid completeUrl: ${completeUrl}`);
+            throw new IllegalArgumentError("Invalid completeUrl", completeUrl);
```

Comment:
If you really need/want to verify URL validity, you should consider using the module [valid-url](https://www.npmjs.com/package/valid-url).